### PR TITLE
Fix various edge case bugs in QValidatedLineEdit

### PR DESCRIPTION
**Pull request by luke-jr**
1. Use a CSS selector to avoid changing the background colour of the tooltip.
2. Re-check validity of input when we first set the validator (probably a no-op in practice).
3. Check validity of input when it is set programmatically via `setText` (eg, via the address book). Probably no-op in practice UNTIL merging #15987 or any other PR that adds a warning for valid addresses.

### CODE REVIEW

File: src/qt/qvalidatedlineedit.cpp [hunk 1]
```diff
@@ -28,7 +34,7 @@ void QValidatedLineEdit::setValid(bool _valid)
     }
     else
     {
-        setStyleSheet(STYLE_INVALID);
+        setStyleSheet("QValidatedLineEdit { " STYLE_INVALID "}");
```
Comment (hebasto) [inline src/qt/qvalidatedlineedit.cpp]: Are quotes around `STYLE_INVALID` macro really needed?

File: src/qt/qvalidatedlineedit.cpp [hunk 2]
```diff
@@ -106,6 +112,7 @@ void QValidatedLineEdit::checkValidity()
 void QValidatedLineEdit::setCheckValidator(const QValidator *v)
 {
     checkValidator = v;
+    checkValidity();
```
Comment (hebasto) [inline src/qt/qvalidatedlineedit.cpp]: This call might cause a side effect that is considered unwanted, no?

Here is an [analogous code:](https://code.woboq.org/qt5/qtbase/src/widgets/widgets/qlineedit.cpp.html#_ZN9QLineEdit12setValidatorEPK10QValidator):
```
void QLineEdit::setValidator(const QValidator *v)
{
    Q_D(QLineEdit);
    d->control->setValidator(v);
}
```
I'd leave `setCheckValidator()` function untouched.

**Review by hebasto — COMMENTED** [review]
Concept ACK aeb18b665c616c3326671b4c7e9d6421306564f0

File: src/qt/qvalidatedlineedit.cpp (hunk 1 above)
Comment (luke-jr) [reply src/qt/qvalidatedlineedit.cpp]: What?

File: src/qt/qvalidatedlineedit.cpp (hunk 2 above)
Comment (luke-jr) [reply src/qt/qvalidatedlineedit.cpp]: I don't know why it would be unwanted.

File: src/qt/qvalidatedlineedit.cpp (hunk 1 above)
Comment (hebasto) [reply src/qt/qvalidatedlineedit.cpp]: Oh, misreading. Sorry for noise.

File: src/qt/qvalidatedlineedit.cpp (hunk 2 above)
Comment (promag) [reply src/qt/qvalidatedlineedit.cpp]: b1a544be109d336c0b53722e3f8b51687972c94e

@hebasto don't understand what side effect you are talking about nor why you point to that code, `setValidator()` is not called.

File: src/qt/qvalidatedlineedit.h [hunk 3]
```diff
@@ -29,6 +29,7 @@ class QValidatedLineEdit : public QLineEdit
     const QValidator *checkValidator;
 
 public Q_SLOTS:
+    void setText(const QString&);
```
Comment (promag) [inline src/qt/qvalidatedlineedit.h]: aeb18b665c616c3326671b4c7e9d6421306564f0

Maybe drop this and add in the constructor:
```cpp
connect(this, &QValidatedLineEdit::textChanged, this, &QValidatedLineEdit::checkValidity);
```

**Review by promag — COMMENTED** [review]
Concept ACK.

File: src/qt/qvalidatedlineedit.h (hunk 3 above)
Comment (luke-jr) [reply src/qt/qvalidatedlineedit.h:32]: Then it will run `checkValidity` twice for every keystroke?

**Comment by fanquake** [comment]
A couple of Concept ACKs here, but no other comments for 6 months. Also seems that the 3rd commit is still a no-op (along with probably the second commit?), and #15987 has now been closed. I'm going to suggest re-opening in the [GUI repo](https://github.com/bitcoin-core/gui).

**Pull request closed without merging**